plugins/modules/sonic_dcbx.py

```diff
@@ -0,0 +1,466 @@
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select pfc
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+
+
+  - name: Modify DCBx Interface configurations
+    dellemc.enterprise_sonic.sonic_dcbx:
+      config:
+        interfaces:
+          - name: Ethernet0
+            enabled: True
+            pfc_tlv_enabled: True
+            ets_configuration_tlv_enabled: True
+            ets_recommendation_tlv_enabled: True
+      state: merged
+
+# After State:
+# ------------
+# sonic# show running-configuration interface Ethernet0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+
+# Using "merged" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+#
+# sonic# show running-configuration | grep dcbx
+# sonic#
+
+  - name: Modify DCBX configurations
+    dellemc.enterprise_sonic.sonic_dcbx:
+      config:
+        global:
+            enabled: true
+      state: merged
+
+# After State:
+# ------------
+# sonic# show running-configuration | grep dcbx
+# !
+# dcbx enable
+# !
+
+# Using "merged" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+# sonic# show running-configuration interface Ethernet 0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select pfc
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+
+
+  - name: Delete and set default DCBx Interface configurations
+    dellemc.enterprise_sonic.sonic_dcbx:
+      config:
+        interfaces:
+          - name: Ethernet0
+            enabled: False
+            pfc_tlv_enabled: False
+            ets_configuration_tlv_enabled: False
+            ets_recommendation_tlv_enabled: False
+      state: deleted
+
+# After State:
+# ------------
+# sonic# show running-configuration interface Ethernet0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+
+# Using "merged" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+# sonic# show running-configuration interface Ethernet 24
+# !
+# interface Ethernet24
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select pfc
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+
+
+  - name: Delete and set default DCBx Interface configurations
+    dellemc.enterprise_sonic.sonic_dcbx:
+      config:
+        interfaces:
+          - name: Ethernet0
+            pfc_tlv_enabled: False
+      state: deleted
+
+# After State:
+# ------------
+# sonic# show running-configuration interface Etherneti 24
+# !
+# interface Ethernet24
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+
+# Using "replaced" state for DCBx interface configuration
```

Comment:
Please revise this example to show that existing non-default DCBx configuration on the affected interface is implicitly deleted if it is not included in the "replaced" options specified for that interface. For example, show an interface with at least one and up to three of the DCBx options in non-default state and a playbook that explicitly sets a currently default state option to the non-default state. This will show that the other (three) options return to the default state.

Reply:
Do you mean for "overridden"? Because for "replaced", the attributes not set in the playbook should not be changed.

Reply:
added more attributes to "replaced" and when "replaced" is for one of them, the rest will not be changed.

Reply:
That is not correct. The entire configuration for the interface is "replaced" by the options specified for that interface in the playbook. This means that if only one option is specified, the others should be "deleted" by being reset to the default values.